### PR TITLE
fix: name module gender

### PR DESCRIPTION
**Pull request by Shinigami92**

### CODE REVIEW

File: src/name.ts [hunk 1]
```diff
@@ -1,5 +1,83 @@
...
+      values = male;
+      break;
+    default:
+      values = generic;
+      break;
+  }
+
+  if (values == null) {
+    if (female != null && male != null) {
+      values = faker.random.arrayElement([female, male]);
```
Comment (ST-DDT) [inline src/name.ts:72]: Can you add a test for this? Just use some locale without the default fallback.

File: src/name.ts [hunk 2]
```diff
@@ -1,5 +1,83 @@
 import type { Faker } from '.';
 
+export enum Gender {
+  female = 'female',
+  male = 'male',
+}
+
+// TODO christopher 21-03-2022: Remove 0 and 1 in v7
```
Comment (prisis) [inline src/name.ts:8]: Wrong todo :)

File: src/name.ts [hunk 3]
```diff
@@ -1,5 +1,83 @@
...
+ * Normalize gender.
+ *
+ * @param gender Gender.
+ * @returns Normalized gender.
+ */
+function normalizeGender(
+  gender?: GenderType
+): Exclude<GenderType, number> | undefined {
+  if (gender == null || typeof gender === 'string') {
+    // TODO christopher 21-03-2022: Cast can be removed when we set `strict: true`
```
Comment (prisis) [inline src/name.ts:21]: same

File: src/name.ts (hunk 2 above)
Comment (Shinigami92) [inline src/name.ts:8]:
```suggestion
// TODO @Shinigami92 21-03-2022: Remove 0 and 1 in v7
```

File: src/name.ts (hunk 3 above)
Comment (Shinigami92) [inline src/name.ts:21]:
```suggestion
    // TODO @Shinigami92 21-03-2022: Cast can be removed when we set `strict: true`
```

**Comment by JessicaSachs** [comment]
Should we go ahead and add support for the gender non-binary in this PR? You're setting up the types and infrastructure, which is good, but maybe we should add `NonBinary` to the enum and support it?

**Comment by Shinigami92** [comment]
> Should we go ahead and add support for the gender non-binary in this PR? You're setting up the types and infrastructure, which is good, but maybe we should add `NonBinary` to the enum and support it?

I would do that in v6.2
This is just a bug fix for now, features can be added later

**Review by ST-DDT — APPROVED** [review]

**Review by JessicaSachs — APPROVED** [review]
Great first step. We'll tackle Non Binary support in 6.2 then 👍🏻

**Pull request merged**